common/txmgr/inmemory_store.go

```diff
@@ -0,0 +1,1909 @@
+	if ms.chainID.String() != chainID.String() {
+		return nil, fmt.Errorf("find_txes_with_meta_field_by_receipt_block_num: %w", ErrInvalidChainID)
+	}
+
+	filterFn := func(tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) bool {
+		if tx.Meta == nil {
+			return false
+		}
+		meta := map[string]interface{}{}
+		if err := json.Unmarshal(json.RawMessage(*tx.Meta), &meta); err != nil {
+			return false
+		}
+		if _, ok := meta[metaField]; !ok {
+			return false
+		}
+		if tx.TxAttempts == nil || len(tx.TxAttempts) == 0 {
+			return false
+		}
+		attempt := tx.TxAttempts[0]
+		if attempt.Receipts == nil || len(attempt.Receipts) == 0 {
+			return false
+		}
+		if attempt.Receipts[0].GetBlockNumber() == nil {
+			return false
+		}
+
+		return attempt.Receipts[0].GetBlockNumber().Int64() >= blockNum
+	}
+
+	txsLock := sync.Mutex{}
+	txs := []*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	wg := sync.WaitGroup{}
+	ms.addressStatesLock.Lock()
+	defer ms.addressStatesLock.Unlock()
+	for _, as := range ms.addressStates {
+		wg.Add(1)
+		go func(as *AddressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) {
+			for _, tx := range as.FetchTxs(nil, filterFn) {
+				etx := ms.deepCopyTx(tx)
+				txsLock.Lock()
+				txs = append(txs, etx)
+				txsLock.Unlock()
+			}
+			wg.Done()
+		}(as)
+	}
+	wg.Wait()
+
+	return txs, nil
+}
+
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) FindTxesWithAttemptsAndReceiptsByIdsAndState(ctx context.Context, ids []big.Int, states []txmgrtypes.TxState, chainID *big.Int) (tx []*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error) {
+	if ms.chainID.String() != chainID.String() {
+		return nil, fmt.Errorf("find_txes_with_attempts_and_receipts_by_ids_and_state: %w", ErrInvalidChainID)
+	}
+
+	filterFn := func(tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) bool {
+		return true
+	}
+
+	txIDs := make([]int64, len(ids))
+	for i, id := range ids {
+		txIDs[i] = id.Int64()
+	}
+
+	txsLock := sync.Mutex{}
+	txs := []*txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	wg := sync.WaitGroup{}
+	ms.addressStatesLock.Lock()
+	defer ms.addressStatesLock.Unlock()
+	for _, as := range ms.addressStates {
+		wg.Add(1)
+		go func(as *AddressState[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) {
+			for _, tx := range as.FetchTxs(states, filterFn, txIDs...) {
+				etx := ms.deepCopyTx(tx)
+				txsLock.Lock()
+				txs = append(txs, etx)
+				txsLock.Unlock()
+			}
+			wg.Done()
+		}(as)
+	}
+	wg.Wait()
+
+	return txs, nil
+}
+
+func (ms *InMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) PruneUnstartedTxQueue(ctx context.Context, queueSize uint32, subject uuid.UUID) (int64, error) {
```

Comment:
So is this method doing the same thing that the evm_tx_store's method does to the persistent store?
Minor concern is that we need to ensure the logic of pruning on DB is same as on this in-memory store.

Just thinking out aloud, if we can change the behavior, to find out all Txs that are to be pruned in-memory, and once we get the list, just call delete on the persistent store on those rows?